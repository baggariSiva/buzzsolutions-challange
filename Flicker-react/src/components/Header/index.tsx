import React, { useState } from "react";

import FlickrIcon from "../../icons/FlickrIcon";
import FlickrLogo from "../../icons/FlickrLogo";
import SearchIcon from "../../icons/SearchIcon";

type Props = {
  setSearchQuery: (searchQuery: string) => void;
};

function Header({ setSearchQuery }: Props): JSX.Element {
  const [inputText, setInputText] = useState<string>("");

  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    event.preventDefault();
    setInputText(event.target.value);
  };

  const handleOnSearch = () => {
    setSearchQuery(inputText);
  };

  return (
    <header className="p-3 text-bg-dark header" data-testid='header'>
      <div className="d-flex flex-wrap align-items-center justify-content-center justify-content-lg-start">
        <a
          href="/"
          className="d-flex align-items-center mb-2 mb-lg-0 text-white text-decoration-none"
        >
          <FlickrIcon />
          <FlickrLogo />
        </a>
        <div className="d-flex search-bar">
          <input
            className="form-control me-2 search-input"
            type="search"
            placeholder="Search"
            aria-label="Search"
            onChange={handleOnChange}
            value={inputText}
          />
          <button
            className="btn btn-outline-primary my-2 my-sm-0"
            onClick={handleOnSearch}
            type="submit"
          >
            <SearchIcon />
          </button>
        </div>
      </div>
    </header>
  );
}

export default Header;
