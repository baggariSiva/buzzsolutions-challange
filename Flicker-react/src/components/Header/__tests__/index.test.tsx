import { render, fireEvent, screen } from "@testing-library/react";
import Header from "..";

describe("Header", () => {
  it("renders without crashing", () => {
    const setSearchQuery = jest.fn();
    render(<Header setSearchQuery={setSearchQuery} />);
    expect(screen.getByTestId("header")).toBeInTheDocument();
  });

  it("updates input value when typing", () => {
    const setSearchQuery = jest.fn();
    render(<Header setSearchQuery={setSearchQuery} />);
    const input = screen.getByPlaceholderText("Search") as HTMLInputElement;
    fireEvent.change(input, { target: { value: "test" } });
    expect(input.value).toBe("test");
  });

  it("calls setSearchQuery when search button is pressed", () => {
    const setSearchQuery = jest.fn();
    render(<Header setSearchQuery={setSearchQuery} />);
    const input = screen.getByPlaceholderText("Search");
    fireEvent.change(input, { target: { value: "test" } });
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(setSearchQuery).toHaveBeenCalledWith("test");
  });

  it("matches snapshot", () => {
    const setSearchQuery = jest.fn();
    const { asFragment } = render(<Header setSearchQuery={setSearchQuery} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
