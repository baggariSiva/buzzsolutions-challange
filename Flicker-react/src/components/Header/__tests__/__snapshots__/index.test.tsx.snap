// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="p-3 text-bg-dark header"
    data-testid="header"
  >
    <div
      class="d-flex flex-wrap align-items-center justify-content-center justify-content-lg-start"
    >
      <a
        class="d-flex align-items-center mb-2 mb-lg-0 text-white text-decoration-none"
        href="/"
      >
        <svg
          fill="none"
          height="40"
          id="flickr"
          viewBox="0 0 48 48"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 24C0 10.7452 10.7452 0 24 0C37.2548 0 48 10.7452 48 24C48 37.2548 37.2548 48 24 48C10.7452 48 0 37.2548 0 24Z"
            fill="#fff"
          />
          <path
            d="M38.2954 23.7986C38.2954 27.2222 35.4692 29.9988 31.9766 29.9988C28.4805 29.9988 25.6527 27.2222 25.6527 23.7986C25.6527 20.3766 28.4805 17.6 31.9766 17.6C35.4692 17.6 38.2954 20.3766 38.2954 23.7986"
            fill="#FF007F"
          />
          <path
            d="M22.241 23.7986C22.241 27.2222 19.4132 29.9988 15.9171 29.9988C12.4262 29.9988 9.6 27.2222 9.6 23.7986C9.6 20.3766 12.4262 17.6 15.9171 17.6C19.4132 17.6 22.241 20.3766 22.241 23.7986"
            fill="#0960D5"
          />
        </svg>
        <svg
          class="flickr-logo"
          id="flickr"
          viewBox="0 0 514.168 155.721"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.591 73.837H0V48.486h21.173V42.05C21.173 10.866 35.353 0 63.324 0c7.577 0 13.404 1.207 17.871 2.013L79.06 26.96c-2.912-1.007-5.827-1.61-11.07-1.61-8.935 0-12.432 6.438-12.432 16.7v6.437h24.862v25.351H55.557v79.467H20.59V73.837zM98.676 2.415h34.964v150.888H98.676zM163.553 4.83h34.964v26.556h-34.964zM163.553 48.486h34.964v104.817h-34.964zM303.604 78.462c-6.41-4.022-12.82-5.833-21.173-5.833-15.152 0-26.808 10.864-26.808 28.971 0 16.7 13.6 27.562 28.749 27.562 7.965 0 15.93-1.81 21.756-5.03l.777 26.96c-8.742 3.02-19.23 4.628-28.554 4.628-33.412 0-58.857-20.118-58.857-54.723 0-34.806 25.445-54.924 58.857-54.924 10.49 0 19.811 1.81 27.777 5.834l-2.524 26.555zM321.477 2.415h34.963v90.132h.388l29.137-44.061h38.266L389.073 96.57l38.073 56.734h-42.347l-27.97-51.905h-.389v51.905h-34.963z"
            fill="#0063db"
          />
          <path
            d="M512.418 77.456c-3.886-1.205-7.772-1.205-11.85-1.205-16.314 0-25.445 12.272-25.445 32.793v44.26h-34.964V48.485h31.856V67.8h.389c6.02-13.278 14.762-21.727 29.914-21.727 4.076 0 8.35.602 11.85 1.205l-1.75 30.178z"
            fill="#ff0084"
          />
        </svg>
      </a>
      <div
        class="d-flex search-bar"
      >
        <input
          aria-label="Search"
          class="form-control me-2 search-input"
          placeholder="Search"
          type="search"
          value=""
        />
        <button
          class="btn btn-outline-primary my-2 my-sm-0"
          type="submit"
        >
          <svg
            class="bi bi-search"
            fill="white"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.742 9.344a5.5 5.5 0 1 0-1.397 1.398l3.86 3.86a1 1 0 0 0 1.415-1.413l-3.858-3.845z"
            />
            <path
              d="M11.5 6a3.5 3.5 0 1 1-7 0 3.5 3.5 0 0 1 7 0z"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
